github-event-service/src/push.ts

```diff
@@ -0,0 +1,59 @@
+import * as fetch from 'node-fetch';
+import {safeLoad as parseYaml} from 'js-yaml';
+
+import {getFirebaseInstance} from './firebase-init';
+
+/** The settings object for the branch manager. */
+export interface BranchManagerSettings {
+  enabled: boolean;
+  branches: Array<{
+    branch: string;
+    label: string;
+  }>;
+};
+
+/** 
+ * Limited implementation of the Github push webhook event from
+ * https://developer.github.com/webhooks/#payloads
+ */
+export interface GithubPushEvent { 
+  ref: string;
+  repository: {
+    id: number;
+    full_name: string;
+  }
+};
+
+/** The Firebase app. */
+const firestore = getFirebaseInstance().firestore();
+
+/** Handle webhook events from github of the type, `push`. */
+export function handlePushEvent(event: GithubPushEvent) {
+  if (event.ref === 'refs/tags/master') {
+    updateRepoSettings(event);
+  }
+  // TODO(josephperrott): Handle pushes to non-master branches.
+}
+
+/** 
+ * Update the settings object in firestore for the repo from the event if the
+ * settings have changed.
+ */
+export async function updateRepoSettings (event: GithubPushEvent) {
+  /** 
+   * The document from the repo settings collected for the repo, as identified
+   * by githubs repo id.
+   */
+  const settingsDoc = firestore.collection('repo_settings').doc(`${event.repository.id}`);
+  /** Firestore ref for the instance of the settings doc. */
+  const settingsRef = await settingsDoc.get();
+  /** The url for the raw settings file from the repo. */
+  const url = `https://raw.githubusercontent.com/${event.repository.full_name}/master/.github/angular-robot.yml`;
+  /** The settings object loaded from the repo via github. */
+  const newSettings: BranchManagerSettings = await fetch(url, {}).then(result => result.text()).then(parseYaml);
```

Comment:
Break the `url` and `fetch` into a separate function like `getRepoSettings(repo)`?

Reply:
I opted to have it all in a single functions since everything within the function will not be needed/used by any other methods/functions

Reply:
I still think it's a bit cleaner with another function